First Commit

cs-sujeet-ahirrao opened the pull request:
First Commit

cs-sujeet-ahirrao committed:
First Commit

cs-pooja-singh commented on taxii2-feed/operations.py:
```diff
@@ -0,0 +1,274 @@
+        return None
+
+
+def tlp(TLP_AMBER, TLP_RED, TLP_WHITE, TLP_GREEN, params):
```
why upper case for function params?

cs-pooja-singh replied:
also, is this function used? don't see its usage anywhere

cs-sujeet-ahirrao replied:
Yes, mam, That is not more used ... deleted

cs-pooja-singh commented on taxii2-feed/operations.py:
```diff
@@ -0,0 +1,274 @@
+        "recordTags": ioc["indicator_types"] if "indicator_types" in ioc else ioc['labels'],
+        "name": ioc["name"],
+        "description": ioc["description"] if "description" in ioc else None,
+        "indicators": extract_artifacts(data=ioc["pattern"]),
```
is this expected to create an m2m with these indicators?

cs-pooja-singh commented on taxii2-feed/operations.py:
```diff
@@ -0,0 +1,274 @@
+        "description": ioc["description"] if "description" in ioc else None,
+        "indicators": extract_artifacts(data=ioc["pattern"]),
+        "valid_from": get_epoch(ioc["valid_from"]),
+        "confidence": params.get("confidence") if params.get("confidence") is not None and params.get(
```
leave it as blank instead of setting 0 if not specified from source

cs-pooja-singh commented on taxii2-feed/operations.py:
```diff
@@ -0,0 +1,274 @@
+        "valid_from": get_epoch(ioc["valid_from"]),
+        "confidence": params.get("confidence") if params.get("confidence") is not None and params.get(
+            "confidence") != '' else 0,
+        "reputation": REPUTATION_MAP.get(params.get("reputation")) if params.get(
```
leave it as blank instead of setting Suspicious if not specified from source

cs-pooja-singh commented on taxii2-feed/constants.py:
```diff
@@ -0,0 +1,10 @@
+TLP_MAP = {'Red': '/api/3/picklists/0472d368-bd15-4f52-a119-d403470cbe43',
+           'Amber': '/api/3/picklists/7bff95b7-6438-4b01-b23a-0fe8cb5b33d3',
+           'Green': '/api/3/picklists/47004ad3-721e-43e0-b729-2ad8ee6441c0',
+           'White': ''}
```
should map to None. setting and empty value might fail during insertion

cs-pooja-singh commented on taxii2-feed/operations.py:
```diff
@@ -0,0 +1,274 @@
+                break
+            for ioc in bundle["objects"]:
+                if ioc["type"] == "indicator" and "spec_version" in ioc.keys():
+                    response.append(stix_spec(ioc, "2.1", params))
```
why do we need spec version in response

cs-sujeet-ahirrao replied:
format of 2.0 and 2.1 is different that's why I added that one

cs-pooja-singh commented on taxii2-feed/connector.py:
```diff
@@ -0,0 +1,19 @@
+
+from .operations import operations, _check_health
+
+logger = get_logger('taxii2_feed')
```
can we rename the connector to taxii2-threat-intel-feed

cs-pooja-singh commented on taxii2-feed/info.json:
```diff
@@ -0,0 +1,319 @@
+        "onchange": {
+          "Basic Authentication": [
+            {
+              "title": "User Name / API Key",
```
"User Name/API Key"

cs-pooja-singh commented on taxii2-feed/info.json:
```diff
@@ -0,0 +1,319 @@
+            }
+          ]
+        },
+        "tooltip": "Time stamp of when the indicator was last seen in STAXX. Date can be specified in Date format or Relative Time Unit format.",
```
tooltip and description is for some other key?

cs-pooja-singh commented on taxii2-feed/info.json:
```diff
@@ -0,0 +1,319 @@
+          "description": "Filter the result set to only include the items added after the specified time."
+        },
+        {
+          "title": "Save Response in File",
```
to  file

cs-sujeet-ahirrao committed:
Renamed connector name and added ingestion playbook

cs-sujeet-ahirrao committed:
Modified Playbook name and var name

1. ingestion 'Create' playbook connector name is modified
2. Changed Taxii to 'taxii' or 'TAXII' everywhere
